### DISCUSSION TITLE
Bilinear or Unified?

### DISCUSSION THREAD

**Discussion by gbeynon99** [Jyers/Marlin#302] [Q&A] [answered]
Which is best? I use the bed vizualiser in Octoprint and need to add in code for bilinear or UBL...

Is the 3x3 HS bilinear?

**Comment by Schnabulation** [comment]
Personally I still use bilinear and `G29` before very print because I want the most recent mesh possible. But I believe in the future UBL is the way to go...

**Comment by Jyers** [comment, marked as answer]
Everything that is not labeled UBL that has a mesh is bilinear. UBL is the more advanced option with the ability to tilt your existing mesh rather than creating a new one. If you want more info check out the wiki page on the different configurations: https://github.com/Jyers/Marlin/wiki/Configuration-Details or Marlin's own documentation: https://marlinfw.org/docs/features/unified_bed_leveling.html
